lsp/server/src/utils/orgUtils.ts

```diff
@@ -116,18 +76,26 @@ export class OrgUtils {
         }
     }
 
-    // Updates the auth state async
+    // Update the auth state async
     private static async checkAuthStatus(): Promise<AuthStatus> {
         if (this.authStatus !== AuthStatus.UNKNOWN) {
             return this.authStatus;
         }
+        try {
+            this.orgName = await this.getDefaultOrg();
+        } catch (e) {
+            this.orgName = undefined;
+        }
```

Comment:
This looks funny. What if in line 85 you get a valid org name back. And let's say the code goes into the `if` block. Then in line 92 why are you setting it to `undefined`? Another way to ask this question is do you need to preeempt to get org name in line 85? Previous version of this file wasn't doing it. I like to understand the significance.

Reply:
'OrgName' is used to construct 'ObjectInfoFolder'. If  connection is invalid, its value should be set to 'undefined' to avoid dirty state.

Reply:
But org name being used to construct `ObjectInfoFolder` has no relevance here.

Compared your change to the previous version. Org name must have been fetched and assigned to `this.orgName` else where. In your code, instead, you decided to do it in `checkAuthStatus()`. Why?

And I go back to my question again. Suppose there was valid `orgName`. But in line 92 if you don't have connection you set `orgName` to undefined. Line 92 is almost identical to the previous version of the code. But I think that one was wrong too for this part. `CheckAuthStatus()` should only check and not change states, such as `orgName`. This is where this function can be improved/refactored. In another words, `checkAuthStatus()` which is signaling for read, in fact, is also doing write.

Reply:
`OrgName` and `authStatus` are the  org states. In previous version, they are maintained in 'getDefaultOrg' and `checkAuthStatus` separately.  In this version, `orgName` is pulled out from `getDefaultOrg` and put into `checkAuthStatus` as the central place to maintain status. `CheckAuthStatus`'s name might be misleading, but it does state maintaining job.

Reply:
Why did you pull out `getDefaultOrg`? That seems to be a good place to keep the one and only place to manage `orgName` state?

Reply:
There are two reasons to pull out orgName setting from `getDefaultOrg`

1. Centralize the state update within `checkAuthStatus`.  We don't want to have two places to maintain states. What is more, everytime `getObjectInfo` is called, `checkAuthStatus` is invoked at the beginning, which makes sure that state is accurate.
2. `getDefaultOrg` should be only read instead of update. This is the same reasoning behind as you have for `CheckAuthStatus`

Reply:
I think you and I differ on point #2. I don't think `getDefaultOrg` can be only be read only because it is a computed property. And throughout the code `this.orgName` is used for a stashed value.

But you can make `getDefaultOrg` read only if you can get rid of all usages `this.orgName` and instead call `getDefaultOrg` whenever you need the org name. That way no stashed value ever goes stale so you don't have to worry about needing to update `this.orgName` in `checkAuthStatus`.

Is there a cost to not stashing? Brief look at `getDefaultOrg` doesn't look like there is one. Even if the aggregator is making file I/O the amount of data that needs to be read off the disk is minimal that I wouldn't consider it a performance hit.

Reply:
@sfdctaka centralized the org name, connection, status into new object OrgInfo , take a look and see if it address your concern.